pkg/resources/api/resource_component.go

```diff
@@ -0,0 +1,238 @@
+package api
+
+import (
+	"reflect"
+
+	tfjson "github.com/hashicorp/terraform-json"
+	"github.com/mohae/deepcopy"
+	"github.com/seal-io/utils/json"
+	"k8s.io/klog/v2"
+)
+
+type ResourceComponentTerraformChange struct {
```

Comment:
How do we use it? I don't find references

Reply:
As we store `[]bytes` in  ComponentChanges, this is used for parse ComponentChanges. It will be used for plan stage that not implemented in this PR.